src/hotspot/cpu/ppc/macroAssembler_ppc.hpp

```diff
@@ -359,11 +359,14 @@ class MacroAssembler: public Assembler {
   address call_c(Register function_entry);
   // For tail calls: only branch, don't link, so callee returns to caller of this function.
   address call_c_and_return_to_caller(Register function_entry);
-  address call_c(address function_entry, relocInfo::relocType rt);
+  address call_c(address function_entry, relocInfo::relocType rt = relocInfo::none);
 #else
   // Call a C function via a function descriptor and use full C
   // calling conventions. Updates and returns _last_calls_return_pc.
   address call_c(Register function_descriptor);
+  address call_c(address function_entry, relocInfo::relocType rt = relocInfo::none) {
+    return call_c((FunctionDescriptor*)function_entry, rt);
```

Comment:
This breaks ABIv1. Please use a cast to `const FunctionDescriptor*` and move your new function below ` address call_c(const FunctionDescriptor* function_descriptor, relocInfo::relocType rt);`.